src/utils/pattern.ts

```diff
@@ -12,7 +12,7 @@ const COMMON_GLOB_SYMBOLS_RE = /[*?]|^!/;
 const REGEX_CHARACTER_CLASS_SYMBOLS_RE = /\[[^[]*]/;
 const REGEX_GROUP_SYMBOLS_RE = /(?:^|[^!*+?@])\([^(]*\|[^|]*\)/;
 const GLOB_EXTENSION_SYMBOLS_RE = /[!*+?@]\([^(]*\)/;
-const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.)[^,.{]*}/;
+const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.).*[^,.{]*}/;
```

Comment:
According to the CodeQL report, with this change we are returning the polynominal regex. I propose to exclude extra characters from the second part of the expression.

```suggestion
const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.)[^{]*}/;
```

Reply:
Done! Thanks!